community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -424,10 +445,43 @@ private void writeState( PagedFile pagedFile ) throws IOException
                     root.id(), root.generation(),
                     freeList.lastId(), freeList.writePageId(), freeList.readPageId(),
                     freeList.writePos(), freeList.readPos() );
+
+            // Write/carry over header
```

Comment:
Nice! This can easily be extracted into a helper method for improved readability.